### PR TITLE
Implement forfeit

### PR DESCRIPTION
**Pull request by smlovin2**
- Adds forfeit method to the game model in game.rb which sets updates the winning_player_id attribute for the game to be the other player who did not foreit
- Adds a forfeit action to the games controller that calls the forfeit method
- Adds a forfeit button to the game show page
- Adds a new attribute winning_player_id to the game record
- Combines users and players so there is just a user model to cover both
- Removes the database_cleaner.rb file since the config stuff wasn't working correctly from the support folder so just adds it to rails_helper.rb so now tests aren't all breaking
- Adds a couple tests for the forfeit method

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class Game < ActiveRecord::Base
   has_many :pieces
-  has_one  :white_player, class_name: 'User'
-  has_one  :black_player, class_name: 'User'
+  belongs_to  :white_player, class_name: 'User'
+  belongs_to  :black_player, class_name: 'User'
+  belongs_to  :winning_player, class_name: 'User'
```
Comment (smlovin2) [inline app/models/game.rb:5]: Do we need this or maybe not so much?

File: app/models/game.rb [hunk 2]
```diff
@@ -22,4 +23,12 @@ def check?
 
     white_king.in_check? || black_king.in_check?
   end
+
+  def forfeit(forfeiting_user)
+    update_attributes(winning_player_id: other_player(forfeiting_user))
```
Comment (smlovin2) [inline app/models/game.rb]: Also wasn't sure if this was the right way to handle this or not

**Comment by smlovin2** [comment]
@mscoutermarsh  Wasn't sure if we need to do more with the

> This will end the game

part from the card. Right now all it does is set the winning player for the game. Is that enough or should there be more dealing with ending the game. Does the game record need an attribute like `over` or `game_over` or is that not really necessary since we can just check if we have winning player and if we do then we know the game is over? Anyway wasn't sure if there was more that needed to be displayed or something to make it more obvious the game was over and that the forfeit happened or if that would come in later cards.

File: app/models/game.rb (hunk 1 above)
Comment (mscoutermarsh) [inline app/models/game.rb:5]: yes, i like that

File: app/models/game.rb (hunk 2 above)
Comment (mscoutermarsh) [inline app/models/game.rb]: :+1: i like it

**Pull request merged**